In [1]:
import pandas as pd
import Saad
import importlib
importlib.reload(Saad)


   ARREST_KEY       DATE OFNS_DESC   BORO  ARREST_PRECINCT AGE_GROUP  \
0         NaN 2006-01-01       NaN  BRONX              NaN       NaN   
1         NaN 2006-01-01       NaN  BRONX              NaN       NaN   

  PERP_SEX_x PERP_RACE_x        VIC_RACE VIC_AGE_GROUP VIC_SEX PERP_SEX_y  \
0        NaN         NaN           BLACK           <18       M          M   
1        NaN         NaN  WHITE HISPANIC         18-24       M          M   

      PERP_RACE_y  PRECINCT  Complaint Year Number Record Create Date  \
0           BLACK      48.0                    NaN                NaN   
1  WHITE HISPANIC      44.0                    NaN                NaN   

  Offense Description Offense Category Arrest Id  
0                 NaN              NaN       NaN  
1                 NaN              NaN       NaN  
   ARREST_KEY       DATE OFNS_DESC   BORO  ARREST_PRECINCT AGE_GROUP  \
0         NaN 2006-01-01       NaN  BRONX              NaN       NaN   
1         NaN 2006-01-01       NaN

<module 'Saad' from '/Users/saadchadrawala/Documents/UIC 2025 Spring Semester/CS 418/Final Project/Saad.py'>

In [7]:
merged_df = pd.read_csv("merged_nypd_data.csv", low_memory=False)


In [9]:
print("Perp Race:")
print(Saad.get_perp_race_counts(merged_df))

Perp Race:
PERP_RACE_y
BLACK                             11967
WHITE HISPANIC                     2517
UNKNOWN                            1837
BLACK HISPANIC                     1398
WHITE                               298
ASIAN / PACIFIC ISLANDER            170
AMERICAN INDIAN/ALASKAN NATIVE        2
Name: count, dtype: int64


In [11]:
print("Precinct Counts:")
print(Saad.get_precinct_counts(merged_df))

Precinct Counts:
PRECINCT
75.0     1004
73.0      852
46.0      720
47.0      717
44.0      699
         ... 
112.0      19
19.0       16
17.0        7
111.0       4
22.0        1
Name: count, Length: 77, dtype: int64


In [13]:
print("Offense Description Counts:")
print(Saad.get_offense_desc_counts(merged_df))

Offense Description Counts:
OFNS_DESC
ASSAULT 3 & RELATED OFFENSES        38238
PETIT LARCENY                       27107
FELONY ASSAULT                      22606
DANGEROUS DRUGS                     18519
MISCELLANEOUS PENAL LAW             16541
VEHICLE AND TRAFFIC LAWS            13783
OTHER OFFENSES RELATED TO THEFT     12621
ROBBERY                             12020
CRIMINAL MISCHIEF & RELATED OF      11854
GRAND LARCENY                       11804
DANGEROUS WEAPONS                   10780
OFFENSES AGAINST PUBLIC ADMINI       8712
BURGLARY                             6450
SEX CRIMES                           5940
FORGERY                              4961
OFFENSES INVOLVING FRAUD             4611
OFF. AGNST PUB ORD SENSBLTY &        4274
POSSESSION OF STOLEN PROPERTY        4050
OTHER STATE LAWS                     3940
CRIMINAL TRESPASS                    2769
INTOXICATED & IMPAIRED DRIVING       2601
OTHER TRAFFIC INFRACTION             2352
GRAND LARCENY OF MOTOR VEHICLE       2

In [15]:
print("Boro Counts:")
print(Saad.get_boro_counts(merged_df))

Boro Counts:
BORO
BROOKLYN         79430
MANHATTAN        65085
BRONX            64112
QUEENS           59612
STATEN ISLAND    11736
Name: count, dtype: int64


In [17]:
print("Age Group Counts:")
print(Saad.get_age_group_counts(merged_df))

Age Group Counts:
AGE_GROUP
25-44    152014
45-64     51116
18-24     43169
<18        9523
65+        4649
Name: count, dtype: int64


In [19]:
print("Sex Counts:")
print(Saad.get_sex_counts(merged_df))

Sex Counts:
PERP_SEX_y
M    16226
U     1499
F      464
Name: count, dtype: int64


# Exploratory Data Analysis: Merged DATASET
This EDA explores a dataset of arrest records from NYC, including details about arrests, perpetrators, victims, and locations. We highlight key trends and summarize the dataset below.
#### Data Description
The dataset contains the following columns:
**ARREST_KEY:** A unique identifier for each arrest.  **DATE:** The date of the arrest or incident. **OFNS_DESC:** Offense description. **BORO:** The NYC borough where the arrest occurred. **ARREST_PRECINCT:** The precinct where the arrest was made. **AGE_GROUP:** Age group of the perpetrator. **PERP_SEX_x:** Sex of the perpetrator. **PERP_RACE_x:** Race of the perpetrator. **VIC_RACE:** Race of the victim. **VIC_AGE_GROUP:** Age group of the victim. **VIC_SEX:** Sex of the victim. **PERP_SEX_y:** Another column for perpetrator sex.**PERP_RACE_y:** Another column for perpetrator race. **PRECINCT:** Precinct number, likely related to the incident. **Complaint Year Number:** Year of the complaint. **Record Create Date:** Date the record was created. **Offense Description:** Detailed offense description.**Offense Category:** Category of the offense. **Arrest Id:** Another identifier for the arrest.
#### FINDINGS OF MERGED DATA 
##### 1. **Racial Distribution of Crime Perpetrators(PERP_RACE_y)?**
- **BLACK:** 11,967 incidents **WHITE HISPANIC:** 2,517 incidents **UNKNOWN:** 1,837 incidents **BLACK HISPANIC:** 1,398 incidents **WHITE:** 298 incidents **ASIAN / PACIFIC ISLANDER:** 170 incidents **AMERICAN INDIAN/ALASKAN NATIVE:** 2 incidents
##### 2. **Racial Distribution of Crime Victims (VIC_RACE)**
- **BLACK:** 12,335 victims **WHITE HISPANIC:** 3,064 victims **BLACK HISPANIC:** 1,852 victims **WHITE:** 563 victims **ASIAN / PACIFIC ISLANDER:** 316 victims **UNKNOWN:** 51 victims **AMERICAN INDIAN/ALASKAN NATIVE:** 8 victims
##### 3. **Borough Distribution:**
- **Brooklyn (K):** 72,325 incidents **Manhattan (M):** 61,969 incidents **Bronx (B):** 58,521 incidents **Queens (Q):** 56,633 incidents **Staten Island (S):** 11,055 incidents
##### 4. **Age Group Distribution:**
- **25-44 years:** 152,034 incidents **45-64 years:** 51,121 incidents **18-24 years:** 43,174 incidents **Under 18 years:** 9,525 incidents **65+ years:** 4,649 incidents
#### Preliminary Conclusions
- **Victim Profile:** Most victims are male, with **Black** being the most common race among victims in the sample.
- **Perpetrator Profile:** Perpetrators are predominantly **male**, and the most common races among perpetrators are **Black** and **White Hispanic**.